client/platform/desktop/frontend/store/settings.ts

```diff
@@ -66,6 +94,13 @@ async function init() {
   }
   settings.value = settingsValue;
   ipcRenderer.send('update-settings', settings.value);
+
+  /* Populate last acknowledged version */
+  const lastKnownVersion = window.localStorage.getItem(VersionKey);
+  if (lastKnownVersion !== null) {
+    knownVersion.value = lastKnownVersion;
```

Comment:
I may be a little mixed up here following this.  So correct me if I'm wrong.  
1. **`knowVersion` is always retrieved from the `VersionKey` or by default it is null.**  
2. `VersionKey` is only set when you `acknowledgeVersion`.  
3. `acknowledgeVersion` is only triggered in the alert inside of the Vue component by clicking on it
4. `acknowledgeVersion` is inside of a v-alert which require `upgradedVersion` to be truthy
5. `upgradedVersion` is only truthy if `knownVersion` is truthy and it meets the semver requirements

How does the `knownVersion` initially get set or the `VersionKey` localStorage if the value is initially empty?  If it's null shouldn't it be set to the `remote.app.getVersion()`.  I.E.  When a user is coming from a version which has no desktop.currentVersion, how is the value ever set?

Reply:
1. yes
2. yes
3. yes
4. yes
5. yes

And yes, it doesn't work, I was changing state as I wrote this, and the end state I got to is unreachable by this code.

> If it's null shouldn't it be set to the remote.app.getVersion()

Yes.